backend/src/test/java/com/rhyun/backend/security/service/UserServiceTest.java

```diff
@@ -119,4 +120,72 @@ void getLoggedInUserTest_whenLoggedInUserDoesNotExist_thenThrow() {
         verify(authentication, times(1)).getPrincipal();
         verify(securityContext, times(1)).getAuthentication();
     }
+
+    @Test
+    void findUserByIdTest_whenIdExists_thenReturnUser() {
+        // GIVEN
+        AppUser user = new AppUser("1", "user1", "password1", AppUserRole.ADMIN);
+        when(userRepository.findById("1")).thenReturn(Optional.of(user));
+
+        // WHEN
+        UserDto actual = userService.findUserById("1");
+
+        // THEN
+        UserDto expected = new UserDto(user.username(), user.password(), user.role());
+        assertEquals(expected, actual);
+        verify(userRepository, times(1)).findById("1");
+    }
+
+    @Test
+    void findUserByIdTest_whenIdDoesNotExist_thenThrow() {
+        // GIVEN
+        when(userRepository.findById("1")).thenReturn(Optional.empty());
+
+        // WHEN
+        // THEN
+        assertThrows(
+                UserNotFoundException.class,
+                () -> userService.findUserById("1")
+        );
+        verify(userRepository, times(1)).findById("1");
+
+        assertThrows(UsernameNotFoundException.class, () -> userService.findUserByUsername("user1"));
```

Comment:
not sure if i am missing something, but why are you also calling findUserByUsername in this test?

Reply:
Thank you for your comment! It was my mistake :)